### PR TITLE
Add build rule to generate docker image for the servicegraph service, and k8s config to deploy it

### PR DESCRIPTION
**Pull request by ZackButcher**
First attempt at docker/kube-ifiying the service graph. For some reason the docker image version's viz-lite library spits out an error, but works locally. Still tracking down why.

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/283)

### CODE REVIEW

**Comment by ZackButcher** [comment]
Kuat, Jason added y'all since you've got more experience messing with this than I: for some reason the package javascript libs break when deployed, but work fine locally. I also can't seem to get bazel to preserve directory structure in the `pkg_tar` rule: I originally pushed the `viz-lite.js` file into `js/viz` but on the produced image the `viz` directory was dropped.

**Comment by ayj** [comment]
What error do you get? Below is what I saw when I tried your PR.

```
$ bazel run //examples/servicegraph/docker:servicegraph
INFO: Found 1 target...
Target //examples/servicegraph/docker:servicegraph up-to-date:
  bazel-bin/examples/servicegraph/docker/servicegraph-layer.tar
INFO: Elapsed time: 0.226s, Critical Path: 0.00s

INFO: Running command line: bazel-bin/examples/servicegraph/docker/servicegraph
Skipping c81b34a13a4c17431476bf701ea9ccb51f9383d443b4fb5574954b95a1953093, already loaded.
Skipping 87ae9f1eb33563bdb7dbc2809ede1ab541373102e108d4de96ecd270a257b8db, already loaded.
Tagging 87ae9f1eb33563bdb7dbc2809ede1ab541373102e108d4de96ecd270a257b8db as istio/examples/servicegraph/docker:servicegraph

$ docker run -it -p 8088:8088 istio/examples/servicegraph/docker:servicegraph
2017/02/25 05:08:47 Starting servicegraph service at :8088

$ docker exec -it <container ID> bash
$ curl -v localhost:8088
* Rebuilt URL to: localhost:8088/
* Hostname was NOT found in DNS cache
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 8088 (#0)
> GET / HTTP/1.1
> User-Agent: curl/7.35.0
> Host: localhost:8088
> Accept: */*
>
< HTTP/1.1 404 Not Found
< Content-Type: text/plain; charset=utf-8
< X-Content-Type-Options: nosniff
< Date: Sat, 25 Feb 2017 05:09:55 GMT
< Content-Length: 19
<
404 page not found
* Connection #0 to host localhost left intact
```

**Comment by ayj** [comment]
>  I also can't seem to get bazel to preserve directory structure in the pkg_tar rule: I originally pushed the viz-lite.js file into js/viz/viz-lite.js but on the produced image the viz directory was dropped (it was at js/viz-lite.js).

Do you want viz-lite.js at /examples/servicegraph/js/viz/viz-lite.js or /examples/servicegraph/js/viz-lite.js? It looks like your docker build rule specified /examples/servicegraph/js. The following patch would add the extra `viz` subdirectory.

```
diff --git a/examples/servicegraph/js/BUILD b/examples/servicegraph/js/BUILD
index 97f2a25..e64701f 100644
--- a/examples/servicegraph/js/BUILD
+++ b/examples/servicegraph/js/BUILD
@@ -10,7 +10,7 @@ pkg_tar(
         exclude = ["BUILD"],
     ),
     mode = "0755",
-    package_dir = "/examples/servicegraph/js",
+    package_dir = "/examples/servicegraph/js/viz",
     tags = ["manual"],
     visibility = ["//examples/servicegraph/docker:__pkg__"],
 )
```

```
root@76c0c19240cd:/# ls /examples/servicegraph/js/viz/viz-lite.js
/examples/servicegraph/js/viz/viz-lite.js
```

**Comment by douglas-reid** [comment]
@ZackButcher try adding a `strip_prefix = "./",` to the BUILD rule for the `pkg_tar`. That should force the directory structure you desire.

https://bazel.build/versions/master/docs/be/pkg.html#pkg_tar

```
The directory structure from the files is preserved inside the tarball but a prefix path determined by strip_prefix is removed from the directory structure.
```

File: examples/servicegraph/cmd/server/BUILD [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package(default_visibility = ["//visibility:private"])
+package(default_visibility = ["//visibility:public"])
```
Comment (douglas-reid) [inline examples/servicegraph/cmd/server/BUILD]: Is there any reason to make this public?  I don't think we want anything outside of examples/servicegraph to be importing the library here.  Let's keep a small scope, if possible.

File: examples/servicegraph/deploy/kube/servicegraph.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: servicegraph
+    spec:
+      containers:
+      - name: servicegraph
+        image: localhost:5000/servicegraph
```
Comment (douglas-reid) [inline examples/servicegraph/deploy/kube/servicegraph.yaml]: this seems not ideal. can we have this be a gcr.io/istio-testing/servicegraph (or similar)? even if we don't push it to gcr.io or docker-hub, we can still run the build and tag it as if we were going to...

File: examples/servicegraph/docker/BUILD [hunk 3]
```diff
@@ -0,0 +1,35 @@
+load("@bazel_tools//tools/build_defs/pkg:pkg.bzl", "pkg_tar")
```
Comment (douglas-reid) [inline examples/servicegraph/docker/BUILD]: can we add to `mixer/docker/gcloud_build.sh` (or put a new script here) that would: `bazel run  //examples/servicegraph/docker:servicegraph gcr.io/$PROJECT/servicegraph:$DOCKER_TAG` (and also tag with `latest` and push via gcloud)?

File: examples/servicegraph/deploy/kube/servicegraph.yaml (hunk 2 above)
Comment (ZackButcher) [inline examples/servicegraph/deploy/kube/servicegraph.yaml]: Yup, sorry, left over from my local testing.

**Comment by ZackButcher** [comment]
PTAL.

**Pull request merged**